### PR TITLE
List data elements in program notification templates

### PR DESCRIPTION
**Pull request by nicolayr**
Adds the list of data elements assigned to a program to the list of template variables in the program notification editor. Clicking a data element in the list inserts the relevant code in the subject or message template (whichever is currently active), on the form `#{uid}`.

### CODE REVIEW

File: src/EditModel/event-program/notifications/NotificationDialog.js [hunk 1]
```diff
@@ -71,23 +76,33 @@ const steps = [
...
+            connect(mapStateToProps, mapDispatchToProps, undefined, { pure: false }),
             createFieldConfigsFor(
                 'programNotificationTemplate',
                 ['name', 'messageTemplate'],
             ),
         )(class extends Component {
             componentDidMount() {
-                // FIXME: Hack to reposition the dialog when the first step is clicked (https://github.com/callemall/material-ui/issues/5793)
-                setTimeout(() => window.dispatchEvent(new Event('resize')), 600);
+                // Hack to reposition the dialog when the first step is clicked (Material-UI bug)
```
Comment (jenniferarnesen) [inline src/EditModel/event-program/notifications/NotificationDialog.js]: Should we leave the issue number in the code?

File: src/EditModel/event-program/notifications/selectors.js [hunk 2]
```diff
@@ -1,6 +1,15 @@
 import { get, compose, first, __ } from 'lodash/fp';
 
-export const getStageNotifications = ({ programStages, programStageNotifications }) => compose(get(__, programStageNotifications), get('id'), first)(programStages);
+export const getStageNotifications = ({ programStages, programStageNotifications }) =>
+    compose(get(__, programStageNotifications), get('id'), first)(programStages);
+
+export const getProgramStageDataElements = ({ programStages, availableDataElements }) => {
+    const programStageDataElements = programStages[0].programStageDataElements
```
Comment (jenniferarnesen) [inline src/EditModel/event-program/notifications/selectors.js]: Will it always be the first element of the programStages array?

File: src/EditModel/event-program/notifications/NotificationDialog.js (hunk 1 above)
Comment (nicolayr) [reply src/EditModel/event-program/notifications/NotificationDialog.js]: Did you check the link? They basically say they won't fix it ;)

File: src/EditModel/stepper/stepper.js [hunk 3]
```diff
@@ -43,14 +43,14 @@ function isActiveStep(activeStep, step, index) {
  * // <MyStepperComponent activeStep={activeStep.key} />
  * ```
  */
-export const createStepperFromConfig = (stepperConfig, orientation = 'horizontal') => ({ activeStep, stepperClicked }) => {
+export const createStepperFromConfig = (stepperConfig, orientation = 'horizontal') => ({ activeStep, stepperClicked,...props }) => {
```
Comment (jenniferarnesen) [inline src/EditModel/stepper/stepper.js]: missing a space before ...props

File: src/EditModel/event-program/notifications/selectors.js (hunk 2 above)
Comment (nicolayr) [reply src/EditModel/event-program/notifications/selectors.js]: Yes, because event programs always have 1 stage (which is hidden in the UI).

**Pull request merged**